### ISSUE TITLE
adding suggestion to add '-amberlib`?

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cpptraj#134]
So I just want to make the process of installing cpptraj easier for inexperienced user. Ideally, new user should know how to get the best configuration by just looking at the output/error rather reading manual.

So this is the process

* ./configure (for dummy running, this is what I normally do)

* ./configure gnu (for the minimal effort after seeing `Usage: ./configure <OPTIONS> [gnu | intel | pgi | clang | cray]`)

If getting 
```
Checking NetCDF
./configure: line 89: ./testp: No such file or directory
  Error: Unable to compile using:
         g++  -O3 -Wall  -DHASBZ2 -DHASGZ -DBINTRAJ -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64  -o testp testp.cpp -lnetcdf
  Please check your compiler settings or configure flags. Error follows:
  --------------------
testp.cpp:2:20: error: netcdf.h: No such file or directory
testp.cpp: In function ‘int main()’:
testp.cpp:3: error: ‘nc_strerror’ was not declared in this scope
```

Should we add a suggestion message, like 'adding `-amberlib` flag if you are using amber or `--full-help` for further? Just want to the flow of installing smoother.

**Comment by hainm** [comment]
should we add `configure full help` for less typing?

after using `conda` for a while, I like its paring

`conda help`
`conda create`
`conda list`

whenever want to see the full help in cpptraj, I always feel struggle with `--full-help` (actually with anything like `-- -`)

**Comment by rmcgibbo** [comment]
Personally, I find the degree to which `configure`'s behavior differs from the behavior of more common UNIX autotools-based `configure` scripts to be my primary source of confusion in dealing with Amber compile configuration. So I like [POSIX](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html) and [GNU](https://www.gnu.org/prep/standards/html_node/Command_002dLine-Interfaces.html) command line conventions. But perhaps that's just me.

**Comment by swails** [comment]
-1 on this.  conda is not a good comparison here -- there are no packages that would behave this way.  The closest common command-line cousin to conda in the Unix world (outside of Python) is git (and GROMACS 5+ now behaves this way as well).  This is a "subcommand" syntax, where it is helpful to think of `conda help` and `conda create` as completely different programs lying within the same global namespace.  This is a helpful distinction in some cases because you can think of `conda`, `git`, and `gmx` as the "general" program driving a hugely diverse and complex ecosystem.  The UNIX philosophy is to have many small programs do well-defined tasks (as opposed to few, monolithic programs do many diverse tasks).  This is why, for example, listing filesystem contents is done with one command (`ls`) and navigating them is another (`cd`).  `conda`, `git`, and `gmx` marry this idea of what programs should do with the more modern concept of namespaces.  The intent is to grok the command `conda create` as the "create program in conda", which allows the anaconda folks to make a program called "create" without having to worry about whether or not that program name is already taken.  Note this is the same thing done by Python's distutils.  Which is why you get help on the build and install subcommands via `python setup.py install --help` and `python setup.py build --help`.

These commands themselves take `--arguments` -- not simply positional keywords.  For example:

```
swails@batman ~/src/pypy_numpy (master) $ conda create --help
usage: conda create [-h] [-y] [--dry-run] [-f] [--file FILE] [--unknown]
                    [--no-deps] [-m] [--use-index-cache] [--use-local]
                    [--offline] [--no-pin] [-c CHANNEL] [--override-channels]
                    [-n ENVIRONMENT | -p PATH] [-q] [--copy] [--alt-hint]
                    [--json] [--clone ENV] [--no-default-packages]
                    [package_spec [package_spec ...]]
[snip]
swails@batman ~/src/pypy_numpy (master) $ conda help --help
usage: conda help [-h] [COMMAND]
[snip]
...
```

So the positional arguments, which are really subcommands, themselves behave like UNIX programs with the prototypical `--long-name`/`-l` flag convention.  `configure` in this case serves way too specific of a purpose (it creates a single config.h file with compile and link options for building cpptraj) to justify the use of subcommands.

Note that conda, git, and gromacs do not use this convention to save on typing, but rather to improve clarity and reduce cognitive load on people using their very complex program.

> Personally, I find the degree to which configure's behavior differs from the behavior of more common UNIX autotools-based configure scripts to be my primary source of confusion in dealing with Amber compile configuration. So I like POSIX and GNU command line conventions. But perhaps that's just me.

As I was introduced to the UNIX ecosystem through Amber, my confusion went the other way (no other programs behaved the way the one I had experience with did).  But I got over that pretty quickly.  Had I come the way you did, I certainly would've found it somewhat strange and a little confusing... but only a little -- for the layman, it is still quite a simple process (`./configure gnu && make install`).  But anything that makes it *less* POSIX/GNU like will get a -1 from me.  I've personally wanted to adjust Amber's configure to accept `--` in place of `-` on long arguments and to use the `CC`, `CXX`, and `FC` environment variables when the compiler argument is missing, but that's too low on the priority list right now :).

**Comment by swails** [comment]
As for making this easier for users to install -- just tell them to `apt-get install netcdf`.  Seems pretty simple to me.

A prerequisite list of packages would be lapack, blas, netcdf, libbz2, fftw3, and libgz, I think.  This list is smaller than many other packages, but would probably be helpful to add to the readme.  For people installing within Amber, this is not any concern.

**Comment by hainm** [comment]
> apt-get install netcdf

how can I use this in casegroup headnode? :P

**Comment by swails** [comment]
email help. or use conda.

**Comment by swails** [comment]
or use -amberlib, but in general this is no more a barrier to installing cpptraj than requiring gfortran (which is another prereq for Linux, I forgot that one).

The only difference to you is that casegroup *has* gfortran, but not libnetcdf.

**Comment by drroe** [comment]
>  I've personally wanted to adjust Amber's configure to accept -- in place of - on long arguments and to use the CC, CXX, and FC environment variables when the compiler argument is missing,

I think implementing the latter is fine and in fact desirable. 

> Should we add a suggestion message, like 'adding -amberlib flag if you are using amber or --full-help for further? Just want to the flow of installing smoother.

To be fair, the README.md does contain the following:

>For systems without BLAS/LAPACK/ARPACK and/or NETCDF libraries installed, the -amberlib flag can be specified to use the ones already compiled in an AmberTools installation ($AMBERHOME must be set), e.g. './configure -amberlib gnu'. 

I think that perhaps the Installation section can be moved a bit higher and add the list of recommended packages to it as Jason suggested.

**Comment by hainm** [comment]
> I think implementing the latter is fine and in fact desirable.

close this issue. If @drroe wants to make above in todo-list, we should open corresponding issue.

**Comment by drroe** [comment]
This may be now somewhat addressed by #144
